save-frontend/src/main/kotlin/org/cqfn/save/frontend/utils/RequestUtils.kt

```diff
@@ -116,12 +116,11 @@ suspend fun WithRequestStatusContext.post(
 internal fun Component<*, *>.classComponentResponseHandler(
     response: Response
 ) {
-    val isUnauthorized = response.status == 401.toShort()
     // dirty hack to determine whether this component contains `setResponse` in its context.
     // If we add another context with a function, this logic will break.
     val hasResponseContext = this.asDynamic().context is Function<*>
-    if (isUnauthorized || !hasResponseContext) {
-        redirectResponseHandler(response)
+    if (!hasResponseContext) {
```

Comment:
Why do we need `noop` here, while it just returns `Unit`?

Could it be just like this, without else branch?

```suggestion
    if (hasResponseContext) {
        this.unsafeCast<Component<*, *>>().withModalResponseHandler(response)
    }
```

Reply:
Yes, that makes sense, even though we don't have a valid fallback solution now. But anyway, we should always check that all class components have correct context type.